Products/Zuul/routers/device.py

```diff
@@ -1575,7 +1575,8 @@ def addDevice(self, deviceName, deviceClass, title=None,
         deviceByName = facade.getDeviceByName(safeDeviceName)
         if deviceByIp and organizer.getZ('zUsesManageIp', True) \
                 or deviceByName and deviceClass == deviceByName.getDeviceClassName():
-            primaryId = deviceByName.getPrimaryId() if deviceByName.getDeviceClassName() == deviceClass else deviceByIp.getPrimaryId()
+            primaryId = deviceByName.getPrimaryId() if isinstance(deviceByName, Device) \
+                    and deviceByName.getDeviceClassName() == deviceClass else deviceByIp.getPrimaryId()
```

Comment:
The `if` statement allows `deviceByIp` to have a `None` value when `deviceByName` is a valid device object.  The `deviceByIp.getPrimaryId()` call should also be protected with a check for validity as well.